pkg/utils/build/source_image.go

```diff
@@ -120,42 +127,122 @@ func IsSourceFilesExistsInSourceImage(srcImage string, gitUrl string, isHermetic
 	return true, nil
 }
 
-func IsAppSourceFilesExists(absExtraSourceDirPath string, gitUrl string) (bool, error) {
+// doAppSourceFilesExist checks if there is app source archive included.
+// For the builds based on Konflux image, multiple app sources could be included.
+func doAppSourceFilesExist(absExtraSourceDirPath string) (bool, error) {
 	//Get the file list from extra_src_dir
 	fileNames, err := utils.GetFileNamesFromDir(absExtraSourceDirPath)
 	if err != nil {
 		return false, fmt.Errorf("error while getting files: %v", err)
 	}
 
 	//Get the component source with pattern <repo-name>-<git-sha>.tar.gz
-	repoName := utils.GetRepoName(gitUrl)
-	filePatternToFind := repoName + "-" + shaValueRegex + tarGzFileRegex
+	filePatternToFind := "^.+-" + shaValueRegex + tarGzFileRegex
 	resultFiles := utils.FilterSliceUsingPattern(filePatternToFind, fileNames)
 	if len(resultFiles) == 0 {
 		return false, fmt.Errorf("did not found the component source inside extra_src_dir, files found are: %v", fileNames)
 	}
-	sourceGzTarFileName := resultFiles[0]
 
-	//Untar the <repo-name>-<git-sha>.tar.gz file
-	err = utils.Untar(absExtraSourceDirPath, filepath.Join(absExtraSourceDirPath, sourceGzTarFileName))
+	fmt.Println("file names:", fileNames)
+	fmt.Println("app sources:", resultFiles)
+
+	for _, sourceGzTarFileName := range resultFiles {
+		//Untar the <repo-name>-<git-sha>.tar.gz file
+		err = utils.Untar(absExtraSourceDirPath, filepath.Join(absExtraSourceDirPath, sourceGzTarFileName))
+		if err != nil {
+			return false, fmt.Errorf("error while untaring %s: %v", sourceGzTarFileName, err)
+		}
+
+		//Get the file list from extra_src_dir/<repo-name>-<sha>
+		sourceGzTarDirName := strings.TrimSuffix(sourceGzTarFileName, ".tar.gz")
+		absSourceGzTarPath := filepath.Join(absExtraSourceDirPath, sourceGzTarDirName)
+		fileNames, err = utils.GetFileNamesFromDir(absSourceGzTarPath)
+		if err != nil {
+			return false, fmt.Errorf("error while getting files from %s: %v", sourceGzTarDirName, err)
+		}
+		if len(fileNames) == 0 {
+			return false, fmt.Errorf("no file found under extra_src_dir/<repo-name>-<git-sha>")
+		}
+	}
+
+	return true, nil
+}
+
+// NewGithubClient creates a GitHub client with custom organization.
+// The token is retrieved in the same way as what SuiteController does.
+func NewGithubClient(organization string) (*github.Github, error) {
```

Comment:
Will it not be possible to reuse [common github client](https://github.com/redhat-appstudio/e2e-tests/blob/ad0dac97d6f6e129736d01f89249ede730780892/pkg/clients/github/client.go#L21) here?

Reply:
This simple wrapper avoids repeating getting the token.